internal/kafka/command_cluster_update.go

```diff
@@ -84,6 +89,41 @@ func (c *clusterCommand) update(cmd *cobra.Command, args []string) error {
 		update.Spec.Config = &cmkv2.CmkV2ClusterSpecUpdateConfigOneOf{CmkV2Dedicated: &cmkv2.CmkV2Dedicated{Kind: "Dedicated", Cku: updatedCku}}
 	}
 
+	if cmd.Flags().Changed("type") {
+		newType, err := cmd.Flags().GetString("type")
+		if err != nil {
+			return err
+		}
+		if newType == "" {
+			return fmt.Errorf("`--type` flag value must not be empty")
+		}
+
+		// Validate cluster type upgrade
+		currentConfig := currentCluster.GetSpec().Config
+		if currentConfig.CmkV2Standard != nil {
+			return fmt.Errorf("cluster type cannot be changed for Standard clusters")
+		}
+		if currentConfig.CmkV2Enterprise != nil {
+			return fmt.Errorf("cluster type cannot be changed for Enterprise clusters")
+		}
+		if currentConfig.CmkV2Dedicated != nil {
+			return fmt.Errorf("cluster type cannot be changed for Dedicated clusters")
+		}
+		if currentConfig.CmkV2Freight != nil {
+			return fmt.Errorf("cluster type cannot be changed for Freight clusters")
+		}
```

Comment:
```suggestion
```

Config is a `OneOf` field, so only one type will be non-nil at a time. So we can remove these blocks and only leave the last block checking basic type. I think that error message is fine for covering these cases too.